flow/flow.yaml

```diff
@@ -25,6 +25,7 @@ components:
             The header is a list of traffic protocol headers.
             The order of traffic protocol headers assigned to the list is the
             order they will appear on the wire.
+            If no packet header is provided Ethernet will be set as default header.
```

Comment:
Disagree with this, the packet list is what is configured, there is also no notation in openapi to explicitly define a default item for the list.

Reply:
@ajbalogh @ankur-sheth won't it be more user-friendly if we configure a default ethernet header under the hood when no packet headers are provided ? not being able to specify default for lists is a limitation in openapi, hence we chose to mention it in description.

Reply:
Reviewed the openapi spec and you can specify a default in this case by setting the minimum array length to 1.  
Be aware that every packet header that gets added now defaults to ethernet instead of custom.  

```yaml
in flow.yaml add minItems to the packet property

        packet:
          description: >-
            The header is a list of traffic protocol headers.
            The order of traffic protocol headers assigned to the list is the
            order they will appear on the wire.
          type: array
          minItems: 1
          items:
            $ref: './packet-headers/header.yaml#/components/schemas/Flow.Header'

in header.yaml change the choice property default to ethernet

    Flow.Header:
      description: >-
        Container for all traffic packet headers
      type: object
      required: [choice]
      properties:
        choice:
          type: string
          enum: [custom, ethernet, vlan, vxlan, ipv4, ipv6, pfcpause, ethernetpause, tcp, udp, gre, gtpv1, gtpv2, arp, icmp, icmpv6, ppp, igmpv1]
          default: ethernet

```

Reply:
Everything else lgtm but I will hold off on my review until this conversation is resolved

Reply:
I have updated the changes according.

Reply:
@shramroy could you please check and answer expected behavior here (I'm trying to understand how minItems works) ?
- user specifies flow without specifying packets
- user specifies flow with packets where list is empty

In both the cases I'm assuming we'll end up with packets containing list of one header, which defaults to ethernet (at least controller needs to ensure it).
If so, the change LGTM.

Reply:
@ashutshkumr yes 3 scenarios covered.